packages/react-ui/src/widgets/legend/LegendWrapper.js

```diff
@@ -150,3 +192,77 @@ function Header({
     </Grid>
   );
 }
+
+const useOpacityControlStyles = makeStyles(({ spacing }) => ({
+  content: {
+    height: 'auto',
+    flex: 1
+  },
+  slider: {
+    marginTop: spacing(1)
+  },
+  input: {
+    width: spacing(8),
+    '& input': {
+      '&[type=number]': {
+        appearance: 'textfield'
+      },
+      '&::-webkit-outer-spin-button': {
+        appearance: 'none',
+        margin: 0
+      },
+      '&::-webkit-inner-spin-button': {
+        appearance: 'none',
+        margin: 0
+      }
+    },
+    '& .MuiInputAdornment-positionEnd': {
+      margin: 0
+    }
+  }
+}));
+
+function OpacityControl({ opacity, onChangeOpacity }) {
+  const classes = useOpacityControlStyles();
+  const handleTextFieldChange = (e) => {
+    const newOpacity = parseInt(e.target.value || '0');
+    onChangeOpacity(Math.max(0, Math.min(100, newOpacity)) / 100);
+  };
+
+  return (
+    <LayerOptionWrapper label='Opacity'>
+      <Box className={classes.content}>
+        <Grid container spacing={2} direction='row' alignItems='center'>
+          <Grid item xs>
+            <Slider
+              value={Math.round(opacity * 100)}
+              min={0}
+              max={100}
+              step={1}
+              onChange={(_, value) => onChangeOpacity(value / 100)}
+              aria-labelledby='input-slider'
+              className={classes.slider}
+            />
+          </Grid>
+          <Grid item>
+            <TextField
+              className={classes.input}
+              value={Math.round(opacity * 100)}
+              margin='dense'
+              onChange={handleTextFieldChange}
+              InputProps={{
+                step: 1,
+                min: 0,
+                max: 100,
+                type: 'number',
+                inputProps: { 'data-testid': 'opacity-slider' },
```

Comment:
We always try to avoid using data-testid as selectors for tests. Can you try with others, following if possible https://testing-library.com/docs/queries/about/#priority ?